### DISCUSSION TITLE
How to create an image that scales the entire page

### DISCUSSION THREAD

**Discussion by hekar** [dolanmiu/docx#853] [Q&A] [answered]
I want to fit an image across the entire page or paragraph section.

How do I do this?

My current approach is to manually figure out how many pixels are used in an A4 document and scale the image accordingly.

**Comment by dolanmiu** [comment, marked as answer]
Yes currently this is the only way

There is no percentage system in place right now unfortunately

**Reply by hekar** [reply]
Thanks man! awesome library btw!
This fits my use case. So it's all good.

There might be a way to calculate all of this in code and use another library to scale the image.

Because we're automatically generating documents it would probably make sense to scale the image ahead of time. Not to waste space.

**Reply by hekar** [reply]
In case anyone needs a library to process images:
https://www.npmjs.com/package/sharp is great.

You can create a `Buffer` and feed it into the `Document`

**Reply by dolanmiu** [reply]
Yup I have used sharp before

I have even once considered importing sharp into the project as a dependency, but that would break the browser compatability of docx
